### PR TITLE
fix: resolved inconsistent exports in ESM

### PR DESCRIPTION
**Pull request by kyranet**
Adds ESM support, for users who prefer using `import` and libraries using canvas that provides ESM bindings.

Thanks for contributing!

- [x] Have you updated CHANGELOG.md?

### CODE REVIEW

File: index.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const { createCanvas } = mod;
+export const { createImageData } = mod;
+export const { loadImage } = mod;
+export const { backends } = mod;
+export const { version } = mod;
+export const { cairoVersion } = mod;
+export const { jpegVersion } = mod;
+export const { gifVersion } = mod;
+export const { freetypeVersion } = mod;
+export const { rsvgVersion } = mod;
```
Comment (VoltrexMaster) [inline index.mjs:28]:
```suggestion
export const {
  Canvas,
  Context2d,
  CanvasRenderingContext2D,
  CanvasGradient,
  CanvasPattern,
  Image,
  ImageData,
  PNGStream,
  DOMMatrix,
  DOMPoint,
  registerFont,
  deregisterAllFonts,
  parseFont,
  createCanvas,
  createImageData,
  loadImage,
  backends,
  version,
  cairoVersion,
  jpegVersion,
  gifVersion,
  freetypeVersion,
  rsvgVersion
} = mod;
```
Nit: Unsure about the style here, these can be narrowed down to a single destructing statement.

**Comment by LinusU** [comment]
As far as I can tell, the library already works perfectly fine to import via ESM, why is this needed?

```sh
$ echo '{"type":"module"}' > package.json

$ npm add canvas

$ cat index.js
import { Canvas } from 'canvas'

console.log(Canvas)

$ node index.js
[Function: Canvas] {
  _registerFont: [Function: _registerFont],
  _deregisterAllFonts: [Function: _deregisterAllFonts]
}
```

**Comment by kyranet** [comment]
Which version of Node.js? I'm trying to import `createCanvas` as well as `loadImage`, but they seem to be nested:

```js
[Module: null prototype] {
  Canvas: [Function: Canvas] {
    _registerFont: [Function: _registerFont],
    _deregisterAllFonts: [Function: _deregisterAllFonts]
  },
  CanvasGradient: [Function: CanvasGradient],
  CanvasRenderingContext2D: [Function: CanvasRenderingContext2D],
  Context2d: [Function: CanvasRenderingContext2D],
  default: {
    Canvas: [Function: Canvas] {
      _registerFont: [Function: _registerFont],
      _deregisterAllFonts: [Function: _deregisterAllFonts]
    },
    Context2d: [Function: CanvasRenderingContext2D],
    CanvasRenderingContext2D: [Function: CanvasRenderingContext2D],
    CanvasGradient: [Function: CanvasGradient],
    CanvasPattern: [Function: CanvasPattern],
    Image: [Function: Image] { MODE_IMAGE: 1, MODE_MIME: 2 },
    ImageData: [Function: ImageData],
    PNGStream: [class PNGStream extends Readable],
    PDFStream: [class PDFStream extends Readable],
    JPEGStream: [class JPEGStream extends Readable],
    DOMMatrix: [class DOMMatrix],
    DOMPoint: [class DOMPoint],
    registerFont: [Function: registerFont],
    deregisterAllFonts: [Function: deregisterAllFonts],
    parseFont: [Function (anonymous)],
    createCanvas: [Function: createCanvas],
    createImageData: [Function: createImageData],
    loadImage: [Function: loadImage],
    backends: {
      ImageBackend: [Function: ImageBackend],
      PdfBackend: [Function: PdfBackend],
      SvgBackend: [Function: SvgBackend]
    },
    version: '2.9.1',
    cairoVersion: '1.17.4',
    jpegVersion: '8',
    gifVersion: '5.2.1',
    freetypeVersion: '2.11.1',
    rsvgVersion: '2.54.0'
  }
}
```

This is in Node.js v16.15.1.

**Comment by kyranet** [comment]
Yeah, that also works, I'll update this PR to do those changes instead.

**Review by LinusU — APPROVED** [review]
Neat 👍

**Pull request merged**